python/app/dialog.py

```diff
@@ -502,88 +312,53 @@ def focus_entity(self):
 
         #################################################################################
         # temp tab handling! Replace with smarter, better solution!
+        # tab handling improved with defining ENTITY_TABS and building the tab widgets
+        # dynamically instead of directly in the .ui file. Once Qt version has been
+        # upgrade to >= 5.15, we can use QTabWidget::setTabVisible instead of clearing
+        # and re-adding the tab widgets each time
 
-        formatter = self._current_location.sg_formatter
-
-        (enabled, caption) = formatter.show_activity_tab
-        self.ui.entity_tab_widget.setTabEnabled(
-            self.ENTITY_TAB_ACTIVITY_STREAM, enabled
-        )
-        self.ui.entity_tab_widget.setTabText(self.ENTITY_TAB_ACTIVITY_STREAM, caption)
-
-        (enabled, caption) = formatter.show_notes_tab
-        self.ui.entity_tab_widget.setTabEnabled(self.ENTITY_TAB_NOTES, enabled)
-        self.ui.entity_tab_widget.setTabText(self.ENTITY_TAB_NOTES, caption)
+        # Block signals emitting on the tab widget to avoid triggering unnecessary data loads
+        self.ui.entity_tab_widget.blockSignals(True)
```

Comment:
Mind doing this from a try/finally to make sure it's set back to `False` even if something goes wrong in between?

Reply:
Definitely